web/src/features/charts/elements/AreaGraphLayers.tsx

```diff
@@ -2,14 +2,18 @@
 /* eslint-disable react/jsx-handler-names */
 import { area, curveStepAfter } from 'd3-shape';
 import { useDarkMode } from 'hooks/theme';
-import React from 'react';
+import React, { useMemo } from 'react';
 import { ElectricityModeType } from 'types';
 import { modeColor } from 'utils/constants';
 
 import { detectHoveredDatapointIndex, getNextDatetime, noop } from '../graphUtils';
 import { SelectedData } from '../OriginChart';
 import { AreaGraphElement } from '../types';
 
+// Function to check if a data point has estimation method
+const hasEstimationMethod = (d: { data: AreaGraphElement }) =>
```

Comment:
This need to check for estimated percentage as well.

And if it just checks for the estimation method it should return a boolean not the estimation method itself.